interfaces/snap_app_set.go

```diff
@@ -197,24 +207,33 @@ func (a *SnapAppSet) Runnables() []Runnable {
 //   - "snap.$snap_instance.{$app1,...$appN, $hook1...$hookN}" if there are
 //     some, but not all, apps/hooks bound
 //   - "snap.$snap_instance.*" if all apps/hooks are bound to the plug or slot
-func labelExpr(apps []*snap.AppInfo, hooks []*snap.HookInfo, snap *snap.Info) string {
+func labelExpr(apps []*snap.AppInfo, hooks []*snap.HookInfo, appSet *SnapAppSet) string {
 	var buf bytes.Buffer
 
 	names := make([]string, 0, len(apps)+len(hooks))
 	for _, app := range apps {
-		names = append(names, app.Name)
+		names = append(names, "."+app.Name)
 	}
+
 	for _, hook := range hooks {
-		names = append(names, fmt.Sprintf("hook.%s", hook.Name))
+		if hook.Component != nil {
+			names = append(names, fmt.Sprintf("+%s.hook.%s", hook.Component.Name, hook.Name))
+		} else {
+			names = append(names, fmt.Sprintf(".hook.%s", hook.Name))
```

Comment:
thought: I wonder if there's a way to rewrite this using Runnables() and some more methods on that?

Reply:
This is done in b2333a5aa187405501608ce7d031f9b6cbdf5e25 and 62b4095d588f36ab81638051da5457e2bf45d067.